addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/KeypadHandler.java

```diff
@@ -8,223 +8,183 @@
  */
 package org.openhab.binding.lutron.handler;
 
-import static org.openhab.binding.lutron.LutronBindingConstants.*;
+import java.util.Arrays;
 
-import org.eclipse.smarthome.core.library.types.OnOffType;
-import org.eclipse.smarthome.core.thing.ChannelUID;
 import org.eclipse.smarthome.core.thing.Thing;
-import org.eclipse.smarthome.core.thing.ThingStatus;
-import org.eclipse.smarthome.core.thing.ThingStatusDetail;
-import org.eclipse.smarthome.core.types.Command;
-import org.openhab.binding.lutron.internal.protocol.LutronCommandType;
+import org.openhab.binding.lutron.internal.KeypadComponent;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Handler responsible for communicating with a keypad.
+ * Handler responsible for communicating with Lutron seeTouch and Hybrid seeTouch keypads used in
+ * RadioRA2 and Homeworks QS systems
  *
- * @author Allan Tong - Initial contribution
+ * @author Bob Adair - Initial contribution
  */
-public class KeypadHandler extends LutronHandler {
-    private static final int COMPONENT_BUTTON1 = 1;
-    private static final int COMPONENT_BUTTON2 = 2;
-    private static final int COMPONENT_BUTTON3 = 3;
-    private static final int COMPONENT_BUTTON4 = 4;
-    private static final int COMPONENT_BUTTON5 = 5;
-    private static final int COMPONENT_BUTTON6 = 6;
-    private static final int COMPONENT_BUTTON7 = 7;
-    private static final int COMPONENT_BUTTONTOPLOWER = 16;
-    private static final int COMPONENT_BUTTONTOPRAISE = 17;
-    private static final int COMPONENT_BUTTONBOTTOMLOWER = 18;
-    private static final int COMPONENT_BUTTONBOTTOMRAISE = 19;
-    private static final int COMPONENT_LED1 = 81;
-    private static final int COMPONENT_LED2 = 82;
-    private static final int COMPONENT_LED3 = 83;
-    private static final int COMPONENT_LED4 = 84;
-    private static final int COMPONENT_LED5 = 85;
-    private static final int COMPONENT_LED6 = 86;
-    private static final int COMPONENT_LED7 = 87;
-
-    private static final Integer ACTION_PRESS = 3;
-    private static final Integer ACTION_RELEASE = 4;
-    private static final Integer LED_STATE = 9;
-
-    private static final Integer LED_OFF = 0;
-    private static final Integer LED_ON = 1;
-
-    private Logger logger = LoggerFactory.getLogger(KeypadHandler.class);
-
-    private int integrationId;
-
-    public KeypadHandler(Thing thing) {
-        super(thing);
-    }
-
-    @Override
-    public void initialize() {
-        Number id = (Number) getThing().getConfiguration().get("integrationId");
-
-        if (id == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No integrationId");
+public class KeypadHandler extends BaseKeypadHandler {
+
+    private static enum COMPONENT implements KeypadComponent {
```

Comment:
Yes, but in this case I capitalized it to emphasize that it is essentially a constant that defines the potential components for a given type of keypad.

Reply:
I agree it is unusual to name enums types all upper cased. They are special classes and can be defined in their own file and passed as method parameter. Because they are so similar to classes they follow the same naming convention.

You can also do `COMPONENT.class.getName();` :wink: 

It would certainly raise a few eyebrows when seeing a `COMPONENT.java` file or something like: 

```java
private void myMethod(COMPONENT component) {

}
```

Reply:
Ok. Fair enough. :-) I'm pushing a new commit that changes this.

Reply:
Thanks! It's also a lot easier on the eyes now. :smile: